javascript/code-challenges/challenges-03.test.js

```diff
@@ -0,0 +1,303 @@
+      baseStat: 65,
+    },
+  ],
+  name: 'snorlax',
+  weight: 4600,
+};
+
+const getBaseStatGreaterThan = (arr, minBaseStat) => {
+  return arr.filter(abilityData => abilityData.baseStat > minBaseStat);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 7 - Stretch Goal
+
+Write a function named getStatName that is an extension of your getBaseStatGreaterThan function from challenge 7. For this function, extend your solution from challenge 7 to only return the name of the stat, rather than the entire stat object.
+
+For example, getStatName(snorlaxData.stats, 50) will return ['special-defense', 'special-attack'].
+------------------------------------------------------------------------------------------------ */
+
+const getStatName = (arr, minBaseStat) => {
+  const viableData = arr.filter(abilityData => abilityData.baseStat > minBaseStat);
+  return viableData.map(viable => viable.stat.name);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 8 - Stretch Goal
+
+Write a function named getCharactersWithoutChildren that, given the array of characters, below, uses filter to return an array of all characters without children.
+------------------------------------------------------------------------------------------------ */
+
+const characters = [
+  {
+    name: 'Eddard',
+    spouse: 'Catelyn',
+    children: ['Robb', 'Sansa', 'Arya', 'Bran', 'Rickon'],
+    house: 'Stark',
+  },
+  {
+    name: 'Jon',
+    spouse: 'Lysa',
+    children: ['Robin'],
+    house: 'Arryn',
+  },
+  {
+    name: 'Cersei',
+    spouse: 'Robert',
+    children: ['Joffrey', 'Myrcella', 'Tommen'],
+    house: 'Lannister',
+  },
+  {
+    name: 'Daenarys',
+    spouse: 'Khal Drogo',
+    children: ['Drogon', 'Rhaegal', 'Viserion'],
+    house: 'Targaryen',
+  },
+  {
+    name: 'Mace',
+    spouse: 'Alerie',
+    children: ['Margaery', 'Loras'],
+    house: 'Tyrell',
+  },
+  {
+    name: 'Sansa',
+    spouse: 'Tyrion',
+    house: 'Stark',
+  },
+  {
+    name: 'Jon',
+    spouse: null,
+    house: 'Snow',
+  },
+];
+
+const getCharactersWithoutChildren = (arr) => {
+  return arr.filter(char => !char.children);
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 9 - Stretch Goal
+
+Write a function named evenOddNumericValues that, given an array as input, uses filter to remove any non-numeric values, then uses map to generate a new array containing the string 'even' or 'odd', depending on the original value.
+
+For example: evenOddNumericValues(['Gregor', 2, 4, 1]) returns ['even', 'even', 'odd'].
+------------------------------------------------------------------------------------------------ */
+
+const evenOddNumericValues = (arr) => {
+  const nums = arr.filter(any => typeof any === 'number');
+  return nums.map(num => num % 2 === 0 && 'even' || 'odd');
```

Comment:
You could also do this with a ternary, which would be my preference.
```js
nums.map(num => num % 2 === 0 ? 'even' : 'odd');
```
and while you're at it, you can chain methods together!
```js
const evenOddNumericValues = arr => arr.filter(num => typeof num === 'number').map(num => num % 2 === 0 ? 'even' : 'odd');
```